compiler/modules/front/codeGen.bal

```diff
@@ -1666,20 +1660,20 @@ function codeGenBitwiseBinaryExpr(CodeGenContext cx, bir:BasicBlock bb, s:Binary
     return { result, block: bb };
 }
 
-function codeGenListConstructor(CodeGenContext cx, bir:BasicBlock bb, Environment env, s:ListConstructorExpr expr) returns CodeGenError|ExprEffect {
+function codeGenListConstructor(CodeGenContext cx, bir:BasicBlock bb, Environment env, Position startPos, s:Expr[] members, Position opPos, t:SemType? expectedType) returns CodeGenError|ExprEffect {
     bir:BasicBlock nextBlock = bb;
     bir:Operand[] operands = [];
-    foreach var member in expr.members {
+    foreach var member in members {
         bir:Operand operand;
         { result: operand, block: nextBlock } = check codeGenExpr(cx, nextBlock, env, member);
         operands.push(operand);
     }
-    t:SemType resultType = <t:SemType>expr.expectedType;
+    t:SemType resultType = <t:SemType>expectedType;
     if t:isEmpty(cx.mod.tc, resultType) {
-        return cx.semanticErr("list not allowed in this context", s:range(expr));
+        return cx.semanticErr("list not allowed in this context", { startPos: node.startPos, endPos: node.endPos });
```

Comment:
@heshanpadmasiri @manuranga  What would be the best way to refactor this line? Should I change `codeGenListConstructor` to pass whole listConstructor as an expr into this function?

Reply:
I think doing this as `return cx.semanticErr("list not allowed in this context",{ startPos, endPos })` (where `endPos` is the endPos of `listConstructor` will be better since it'll highlight the same range.

Reply:
`Node` is just a copy paste bug from me. :), will do the change